tldw_Server_API/app/api/v1/endpoints/embeddings_v5_production_enhanced.py

```
@@ -1217,7 +1219,17 @@ def get_cache_key(
         key_parts.append(backend_identity)
     key_string = "|".join(key_parts)
-    return hashlib.sha256(key_string.encode()).hexdigest()
+    return hmac.new(_embedding_cache_key_secret(), key_string.encode("utf-8"), hashlib.sha256).hexdigest()
+
+
+@lru_cache(maxsize=1)
+def _embedding_cache_key_secret() -> bytes:
+    """Return a stable keyed-hash secret for embedding cache partitioning."""
+    try:
+        return derive_hmac_key()
+    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)